app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb

```diff
@@ -39,6 +39,10 @@ description: |
   end
   options << "locale=#{host_param('lang') || 'en_US'}"
   options = options.join(' ')
+
+  if host_param_true?("serial_console")
+    options.push("console=#{host_param("serial_console_dev", "ttyS0")}")
```

Comment:
I feel this will fail, because one line before this, `join()` the options and thus making them a string.
I think you need to have this a line above we join the options.
Also it would probably look better if we would be consistent here and used `options <<`